### PR TITLE
ATO-981: set is new account in orch session

### PR DESCRIPTION
**Pull request by Ryan-Andrews99**
## What:

- Sets the value of isNewAccount in the Orch session at the same point as it is set in the regular session
- This includes updating:
   -  Authentication Callback Handler to update this using Auth's userinfo response
   - Updates the AuthCodeGenerationService to update the Orch session alongside the redis session
## How to review
- Code review commit-by-commit
- Deploy to orch dev environment
    - Run through a journey 
    - Check all is well
    - Ran auth acceptance tests against deployed env (sandpit), tests passed
 - Once deployed:
     - Run through journey in staging as feature will be enabled 
     - Verify that in the case of a return code this value is set  
     - Check doc app journey
     - Enable feature flag in higher environments in future PRs

## Checklist

- [x] Impact on orch and auth mutual dependencies has been checked.

- [ ] A UCD review has been performed.

### CODE REVIEW

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java [hunk 1]
```diff
@@ -375,8 +375,13 @@ public APIGatewayProxyResponseEvent handleRequest(
 
                 Boolean newAccount = userInfo.getBooleanClaim("new_account");
                 AccountState accountState = newAccount ? AccountState.NEW : AccountState.EXISTING;
+                OrchSessionItem.AccountState OrchAccountState =
+                        newAccount
+                                ? OrchSessionItem.AccountState.NEW
+                                : OrchSessionItem.AccountState.EXISTING;
```
[the remark this answers is not in the document]
Comment (Ryan-Andrews99) [reply oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java:381]: I was thinking a bit about the null safety of the userInfo response from Auth, did attempt a refactor which wrapped it in a record with lots of optional returning methods. Found it was a bit of faff but maybe something we'd want to revisit in the future?

Anyway to answer the question, this **_should_** always be attached (same as above), happy to implement this but would probably do the same thing to the above value too

File: integration-tests/src/test/java/uk/gov/di/authentication/api/AuthenticationCallbackHandlerIntegrationTest.java [hunk 2]
```diff
@@ -1086,4 +1103,10 @@ private void validateClaimsInJar(SignedJWT signedJWT, boolean reproveIdentity)
         assertThat(signedJWT.getJWTClaimsSet().getClaim("scope"), equalTo("openid"));
         assertThat(signedJWT.getHeader().getAlgorithm(), equalTo(ES256));
     }
+
+    private void assertOrchSessionUpdatedWithNewAccountState() {
```
[the remark this answers is not in the document]
Comment (Ryan-Andrews99) [reply integration-tests/src/test/java/uk/gov/di/authentication/api/AuthenticationCallbackHandlerIntegrationTest.java:1107]: Yeah not sure why I went through the effort of making a whole new method 🫤

[the remark this answers is not in the document]
Comment (Ryan-Andrews99) [reply integration-tests/src/test/java/uk/gov/di/authentication/api/AuthenticationCallbackHandlerIntegrationTest.java]: Done!

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (billrdunn) [reply oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java:385]: Or this

```
Boolean newAccount = userInfo.getBooleanClaim("new_account");
AccountState accountState;
if (newAccount == null) {
    accountState = AccountState.UNKNOWN;
} else {
    accountState = newAccount ? AccountState.NEW : AccountState.EXISTING;
}

OrchSessionItem.AccountState orchAccountState;
if (newAccount == null) {
    orchAccountState = OrchSessionItem.AccountState.UNKNOWN;
} else {
    orchAccountState = newAccount ? OrchSessionItem.AccountState.NEW : OrchSessionItem.AccountState.EXISTING;
}
```

[the remark this answers is not in the document]
Comment (Ryan-Andrews99) [reply oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java]: Done!

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java [hunk 3]
```diff
@@ -389,7 +402,7 @@ public APIGatewayProxyResponseEvent handleRequest(
                 } else {
                     sessionService.storeOrUpdateSession(userSession.setNewAccount(accountState));
                 }
-
+                orchSessionService.updateSession(orchSession.withAccountState(orchAccountState));
```
Comment (billrdunn) [inline oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java:405]: Is this meant to be feature flagged?

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/entity/AuthUserInfoClaims.java [hunk 4]
```diff
@@ -15,7 +15,8 @@ public enum AuthUserInfoClaims {
     PHONE_VERIFIED("phone_number_verified"),
     SALT("salt"),
     VERIFIED_MFA_METHOD_TYPE("verified_mfa_method_type"),
-    CURRENT_CREDENTIAL_STRENGTH("current_credential_strength");
+    CURRENT_CREDENTIAL_STRENGTH("current_credential_strength"),
+    NEW_ACCOUNT("new_account");
```
Comment (billrdunn) [inline oidc-api/src/main/java/uk/gov/di/authentication/oidc/entity/AuthUserInfoClaims.java:19]: Worth setting this on the auth-external-api version as well?

Also looks like there are a couple of other usages of `"new_account"` that could be replaced

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java (hunk 3 above)
Comment (Ryan-Andrews99) [reply oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java:405]: Nope as we hit this lambda running through a journey in dev so I can be certain this lambda works

**Review by billrdunn — DISMISSED** [review]

**Pull request merged**